pyccel/codegen/printing/cucode.py

```diff
@@ -0,0 +1,121 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+Provide tools for generating and handling CUDA code.
+This module is designed to interface Pyccel's Abstract Syntax Tree (AST) with CUDA,
+enabling the direct translation of high-level Pyccel expressions into CUDA code.
+"""
+
+from pyccel.codegen.printing.ccode import CCodePrinter, c_imports
+
+from pyccel.ast.variable    import InhomogeneousTupleVariable
+from pyccel.ast.core        import Declare, Import, Module
+
+from pyccel.errors.errors   import Errors
+
+
+import_dict = {'omp_lib' : 'omp' }
+
+c_library_headers = (
+    "complex",
+    "ctype",
+    "float",
+    "math",
+    "stdarg",
+    "stdbool",
+    "stddef",
+    "stdint",
+    "stdio",
+    "stdlib",
+    "string",
+    "tgmath",
+    "inttypes",
+)
+
+errors = Errors()
+
+# TODO: add examples
+
+__all__ = ["CudaCodePrinter"]
+
+class CudaCodePrinter(CCodePrinter):
+    """
+    Print code in CUDA format.
+
+    This printer converts Pyccel's Abstract Syntax Tree (AST) into strings of CUDA code.
+    Navigation through this file utilizes _print_X functions,
+    as is common with all printers.
+
+    Parameters
+    ----------
+    filename : str
+            The name of the file being pyccelised.
+    prefix_module : str
+            A prefix to be added to the name of the module.
+    """
+    printmethod = "_cucode"
+    language = "cuda"
+
+    def __init__(self, filename, prefix_module = None):
+
+        errors.set_target(filename, 'file')
+
+        super().__init__(filename)
+        self.prefix_module = prefix_module
+        self._additional_imports = {'stdlib':c_imports['stdlib']}
+        self._additional_code = ''
+        self._additional_args = []
+        self._temporary_args = []
+        self._current_module = None
+        self._in_header = False
+
+    def _print_Module(self, expr):
+        self.set_scope(expr.scope)
+        self._current_module = expr.name
+        body    = ''.join(self._print(i) for i in expr.body)
+
+        global_variables = ''.join([self._print(d) for d in expr.declarations])
```

Comment:
I'm surprised pylint didn't complain about this. Are we sure that this file is being checked?
```suggestion
        global_variables = ''.join(self._print(d) for d in expr.declarations)
```

Reply:
not sure why will it complain in this case!!

Reply:
I was expecting to see this error:
https://pylint.readthedocs.io/en/latest/user_guide/messages/refactor/use-a-generator.html

My suggestion uses a generator which means that each `self._print(d)` object is created on the fly as it is added to the string, while using a list causes a list to be allocated, filled, then destroyed

Reply:
[ec493e8](https://github.com/pyccel/pyccel-cuda/pull/32/commits/ec493e8fc12e04a7ee41bafc866dfc24ae61e9d7)

Reply:
[c61c5af](https://github.com/pyccel/pyccel-cuda/pull/32/commits/c61c5afbe42713bf16a7a00e068c8ef3567797c2)
[cd4af31](https://github.com/pyccel/pyccel-cuda/pull/32/commits/cd4af311938b22a6924edd10531370d23c1fdbc5)